### ISSUE TITLE
Nginx resolver and IPv6 in container (podman)

### ISSUE THREAD

**Issue by gitouche-sur-osm** [jitsi/docker-jitsi-meet#1444]
Hello

I know podman is [not supported](https://github.com/jitsi/docker-jitsi-meet/issues/201), but we have been running jitsi in podman flawlessly for a long time.  
The fix to [nginx resolver](https://github.com/jitsi/docker-jitsi-meet/commit/53b265455661fe0a6fc986ad589d6ecfd911847c) was very much needed for podman to continue working (thank you!).

After upgrading to 8044, I was surprised to find jitsi-meet still could not resolve the prosody server name in nginx internal resolver.  
I found a new error in Nginx logs : `unexpected A record in DNS response` and figured it was expecting another type of DNS record (AAAA?).

The containers in podman now have full IPv6 support (even when not using it, the containers have a fe80:: Link-local IPv6 address) and it changes nginx behaviour while resolving DNS names.

We resolved this by adding `ipv6=off` to the resolver configuration in https://github.com/jitsi/docker-jitsi-meet/blob/c44c59e6cb5ccdadfe0e710b7c649545fb5904a8/web/rootfs/defaults/nginx.conf#L30

This is the first time we had to alter the image building process to make docker-jitsi-meet work with podman.
Would you consider adding `ipv6=off` to your nginx.conf template file? I don't think it will break any docker configuration.

Thank you

**Comment by saghul** [comment]
An env variable that allows to disable it would be acceptable, while keeping the current behavior by default.

**Comment by gitouche-sur-osm** [comment]
> An env variable that allows to disable it would be acceptable, while keeping the current behavior by default.

That would be perfect. I did some more testing and found no other way to fix the issue.  
I would be glad to try to provide a PR for an env variable. What name should it use?

**Comment by saghul** [comment]
How about DISABLE_IPV6 ?

**Comment by benz0li** [comment]
> How about DISABLE_IPV6 ?

Please don't. There is already `ENABLE_IPV6`, which is used for [web](https://github.com/jitsi/docker-jitsi-meet/blob/d6fac8e0c38294463a535442bba3dd27327b966c/web/rootfs/defaults/default#L4) and [prosody](https://github.com/jitsi/docker-jitsi-meet/blob/b2241315ade699df45052586df0771fc852d265c/prosody/rootfs/defaults/prosody.cfg.lua#L117).

I suggest renaming `ENABLE_IPV6` to `ENABLE_LISTEN_IPV6` (default: "true") and naming the new env variable `DISABLE_RESOLV_IPV6` or `DISABLE_RESOLVE_IPV6` (default: "false").

P.S.: I work around this issue by setting `NGINX_RESOLVER=127.0.0.11 ipv6=off`.

**Comment by saghul** [comment]
Ah I had forgotten about that one, thanks for pointing it out!

Btw, If there is already a way to work around it, not sure we need a new one...

**Comment by gitouche-sur-osm** [comment]
I can't use that, as the container resolver IP address cannot be predicted and is copied over from the local `/etc/resolv.conf` to nginx configuration (see https://github.com/jitsi/docker-jitsi-meet/commit/53b265455661fe0a6fc986ad589d6ecfd911847c).

**Comment by saghul** [comment]
Right. Then yeah we need an extra env variable.

Maybe we can be exploit and call it NGINX_RESOLVER_DISABLE_IPV6

**Comment by benz0li** [comment]
> Maybe we can be exploit and call it NGINX_RESOLVER_DISABLE_IPV6

`NGINX_RESOLVER_DISABLE_IPV6` is perfectly fine, too.

**Comment by gitouche-sur-osm** [comment]
I finally found the root cause for nginx behaviour.  
Following this [comment](https://trac.nginx.org/nginx/ticket/1837#comment:1), I found a bug in the DNS resolver podman uses `"aardvark-dns"` : it answers A records when asked for AAAA records and this is not expected behaviour. Nginx errors out on it.

This got fixed in [aardvark-dns 1.2.0](https://github.com/containers/aardvark-dns/releases/tag/v1.2.0) as `coredns: do not combine results of A and AAAA records`. I am using aardvark-dns 1.1.0 at the moment!

In my case, the needed conditions for the bug to appear are : 
- `podman 4.x` (current) with `netavark+aardvark-dns` network stack (this is the new network stack, default is still `CNI`)
- `RHEL/CentOS 8` and clones. `Fedora` and `RHEL/CentOS 9` already have a fixed `aardvark-dns (1.3.0)`!

Hopefully `RHEL/CentOS 8` will get the news `aardvark-dns 1.3.0` in future release 8.8 (6 months from now).  

Until then, I can live with patching `ipv6=off` on top of the repo to build the images. I'm not sure anyone else will ever encounter the bug, so feel free to close the issue if you think it better. I would use a env. variable if it was there but the root cause is not with `docker-jitsi-meet` :+1:
